extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/SmallRyeReactiveMessagingKafkaProcessor.java

```diff
@@ -46,18 +48,34 @@ public void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
                         .build());
     }
 
+    /**
+     * Handles the serializer/deserializer detection and whether or not the graceful shutdown should be used in dev mode.
+     */
     @BuildStep
-    public void defaultSerdeConfig(ReactiveMessagingKafkaBuildTimeConfig buildTimeConfig,
+    public void defaultChannelConfiguration(
+            LaunchModeBuildItem launchMode,
+            ReactiveMessagingKafkaBuildTimeConfig buildTimeConfig,
+            ReactiveMessagingKafkaConfig runtimeConfig,
             CombinedIndexBuildItem combinedIndex,
             BuildProducer<RunTimeConfigurationDefaultBuildItem> defaultConfigProducer) {
 
-        if (!buildTimeConfig.serializerAutodetectionEnabled) {
-            return;
-        }
-
         DefaultSerdeDiscoveryState discoveryState = new DefaultSerdeDiscoveryState(combinedIndex.getIndex());
+        if (buildTimeConfig.serializerAutodetectionEnabled) {
+            discoverDefaultSerdeConfig(discoveryState, defaultConfigProducer);
+        }
 
-        discoverDefaultSerdeConfig(discoveryState, defaultConfigProducer);
+        if (launchMode.getLaunchMode().isDevOrTest()) {
+            if (!runtimeConfig.enableGracefulShutdownInDevAndTestMode) {
+                for (AnnotationInstance annotation : discoveryState.findAnnotationsOnMethods(DotNames.INCOMING)) {
```

Comment:
You're only looking for `@Incoming` methods, but you can also have `@Channel` injection points that correspond to incoming channels. The body of this loop would also work for the `@Channel` annotations, so replacing this line with the following should be enough:

```java
List<AnnotationInstance> incomings = discoveryState.findAnnotationsOnMethods(DotNames.INCOMING);
List<AnnotationInstance> channels = discoveryState.findAnnotationsOnInjectionPoints(DotNames.CHANNEL);
List<AnnotationInstance> annotations = new ArrayList<>();
annotations.addAll(incomings);
annotations.addAll(channels);
for (AnnotationInstance annotation : annotations) {
```

Reply:
Good point!

Reply:
Fixed